akka-http-core/src/main/scala/akka/http/impl/engine/client/HttpsProxyGraphStage.scala

```diff
@@ -48,7 +49,7 @@ private final class HttpsProxyGraphStage(targetHostName: String, targetPort: Int
 
   override def shape: BidiShape[ByteString, ByteString, ByteString, ByteString] = BidiShape.apply(sslIn, bytesOut, bytesIn, sslOut)
 
-  private val connectMsg = ByteString(s"CONNECT $targetHostName:$targetPort HTTP/1.1\r\nHost: $targetHostName\r\n\r\n")
+  private val connectMsg = ByteString(s"CONNECT $targetHostName:$targetPort HTTP/1.1\r\nHost: $targetHostName\r\n${proxyAuth.fold("")(httpCredentials ⇒ s"${`Proxy-Authorization`(httpCredentials)}\r\n")}\r\n")
```

Comment:
`ModeledHeader` extends from `ToStringRenderable` so `toString` does exactly the right thing in this case. See here: https://github.com/akka/akka-http/blob/1bee8d00277af2bee8c2ed1f1d71c118fc2be0c9/akka-http-core/src/main/scala/akka/http/impl/util/Rendering.scala#L37-L37

Though it would be nice if you put the fold into a `private def` just to make it a bit more readable.